### PR TITLE
Httpd (Apache) add CI for Ubuntu LTS 18 & 20

### PR DESCRIPTION
**Pull request by TomRoSystems**
- moving github actions file to top dir from instrumentation/httpd
- added artifact creation from compiled module

### CODE REVIEW

File: .github/workflows/httpd.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+on:
+  push:
+    branches: "*"
+    paths:
+      - 'instrumentation/httpd/**'
+      - '.github/workflows/httpd.yml'
+  pull_request:
+    branches: [ main ]
+
```
Comment (seemk) [inline .github/workflows/httpd.yml:11]: Should add paths filter here as well

File: .github/workflows/httpd.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+      - name: Setup buildtools
+        run: |
+          sudo ./instrumentation/httpd/setup-buildtools.sh
+          sudo ./instrumentation/httpd/setup-environment.sh
+      - name: Compile
+        run: |
+          cd instrumentation/httpd && ./build.sh
```
Comment (seemk) [inline .github/workflows/httpd.yml:29]: Is it a release build btw?

File: .github/workflows/httpd.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        run: |
+          sudo ./instrumentation/httpd/tools/setup-tools.sh buildifier
+      - name: Check formatting
+        run: ./instrumentation/httpd/tools/format-bazel.sh
+
+  e2e_tests:
+    name: Run end-to-end tests
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
```
Comment (seemk) [inline .github/workflows/httpd.yml:65]: :+1:

File: instrumentation/httpd/src/otel/opentelemetry.cpp [hunk 4]
```diff
@@ -86,6 +86,7 @@ void initTracer()
   }
 
   // add custom-configured resources
+  opentelemetry::sdk::resource::ResourceAttributes resAttrs;
```
Comment (seemk) [inline instrumentation/httpd/src/otel/opentelemetry.cpp:89]: Hmm, how did this compile before?

Comment (TomRoSystems) [reply instrumentation/httpd/src/otel/opentelemetry.cpp:89]: It was not... :(

File: .github/workflows/httpd.yml (hunk 1 above)
Comment (TomRoSystems) [reply .github/workflows/httpd.yml:14]: Added

**Review by seemk — APPROVED** [review]
The failing nginx tests were fixed in #20 so no need to worry about this

**Review by lalitb — APPROVED** [review]

**Comment by lalitb** [comment]
> The failing nginx tests were fixed in #20 so no need to worry about this

Just to confirm, this will be good to merge with the failing cases? Not sure why there is no option to update the branch with main before merging.

**Comment by seemk** [comment]
Indeed, perhaps @TomRoSystems can merge main into this

**Comment by TomRoSystems** [comment]
> Indeed, perhaps @TomRoSystems can merge main into this

Merged

**Pull request merged**